main/eval/src/mill/eval/Evaluator.scala

```diff
@@ -21,7 +21,7 @@ trait Evaluator {
   def outPath: os.Path
   def externalOutPath: os.Path
   def pathsResolver: EvaluatorPathsResolver
-  def workerCache: collection.Map[Segments, (Int, Val)]
+  def workerCache: collection.mutable.Map[Segments, (Int, Val)]
```

Comment:
If this counts as a change due to bincompat, we can create a new method with a new name that's `private[mill]` until 0.13.0 comes and we can change the type of the old method